### PR TITLE
Upgrade build to node 10.16.0 and update dependencies

### PR DESCRIPTION
**Pull request by owenniblock**
### Motivation

Recommended LTS version of node is 10.16.0 - we should be using that.

Also we patch dependencies for security or other issues but need to ensure we keep other dependencies up to date as well.

### Tasks Complete
- [x] Node updated and any breaking changes fixed
- [x] Dependencies updated and any breaking changes fixed
- [x] Unit tests running correctly
- [x] Depreciation messages removed from `npm test`
- [x] Docs updated
- [ ] Manual testing performed

### CODE REVIEW

**Comment by owenniblock** [comment]
Suggest we merge this to master but do the testing (if possible / time allows) before release.

File: src/util/misc/promisfied-https.ts [hunk 1]
```diff
@@ -13,7 +13,11 @@ export async function getToFile(url: string, filePath: string): Promise<void> {
         resolve();
       });
     }).on("error", (err: NodeJS.ErrnoException) => {
-      fs.unlink(filePath);
+      fs.unlink(filePath, function (failed) {
+        if (failed) {
+            console.error(failed);
```
Comment (jonstoneman) [inline src/util/misc/promisfied-https.ts]: indentation

File: src/util/token-store/file/file-token-store.ts [hunk 2]
```diff
@@ -30,7 +31,10 @@ export class FileTokenStore implements TokenStore {
 
   list(): rx.Observable<TokenEntry> {
     this.loadTokenStoreCache();
-    return rx.Observable.from(toPairs(this.tokenStoreCache)).map((pair) => ({ key: pair[0], accessToken: pair[1]}));
+    //return rx.from(toPairs(this.tokenStoreCache)).map((pair) => ({ key: pair[0], accessToken: pair[1]}));
```
Comment (jonstoneman) [inline src/util/token-store/file/file-token-store.ts]: delete commented out code

File: test/commands/codepush/deployment/history-test.ts [hunk 3]
```diff
@@ -76,9 +78,10 @@ describe("CodePush deployment history", () => {
       const result = await command.execute();
 
       const tableRows: string[][] = stubbedOutTable.lastCall.args[1];
-      const unchalkedRows = tableRows.map((row) => row.map((element) => chalk.stripColor(element)));
+      expect(tableRows).to.be.an("array");
+      const unchalkedRows: string[][] = tableRows.map((row) => row.map((element) => stripAnsi(element)));
 
-      // Assert
+      // // Assert
```
Comment (jonstoneman) [inline test/commands/codepush/deployment/history-test.ts]: remove extra `//`

File: contributing.md [hunk 4]
```diff
@@ -48,6 +50,9 @@ gulp is used as a task runner under the hood, but should not be called directly.
 By default, to run the CLI you need to compile the typescript explicitly first. The `ts-node` command line tool will let you run
 `.ts` files directly from the command line. Install it with `npm install -g ts-node`.
 
+The latest `ts-node` required installation in local dev to work
+correctly.
```
Comment (jonstoneman) [inline contributing.md]: seems to be a variety of different line lengths in this file - could this be made more consistent at least in the changes?

**Comment by owenniblock** [comment]
Merging this. We'll test it before we release.

**Pull request merged**